.github/workflows/ci_coverage.yml

```
@@ -92,7 +92,7 @@ jobs:
           ccache -sv
 
       - name: Submit coverage build
-        uses: codecov/codecov-action@v3
+        uses: codecov/codecov-action@v4
         with:
```

```suggestion
        uses: codecov/codecov-action@v4
        env:
            CODECOV_TOKEN: ${{ secrets.CODECOV_TOKEN }}
        with:
```

and follow the directions at https://docs.codecov.com/docs/adding-the-codecov-token

---

And adding `fail_ci_if_error: true` to the `with` section isn't a bad idea either